reconcile/utils/saasherder/saasherder.py

```diff
@@ -688,9 +688,10 @@ def _collect_namespaces(self) -> list[Namespace]:
                             environment=target.namespace.environment,
                             app=target.namespace.app,
                             cluster=target.namespace.cluster,
-                            # managedResourceTypes is defined per saas_file
-                            # add it to each namespace in the current saas_file
+                            # managedResourceTypes and managedResourceNames are defined per saas_file
+                            # add them to each namespace in the current saas_file
                             managed_resource_types=saas_file.managed_resource_types,
+                            managed_resource_names=saas_file.managed_resource_names,
```

Comment:
this is the most important line of this PR. we are essentially overriding the namespace's `managedResourceNames` with what is in the saas file. this is the same approach as `managedResourceTypes`, as can be seen one line above.